### DISCUSSION TITLE
Single helm command installation for linkerd2 and linkerd-viz components

### DISCUSSION THREAD

**Discussion by shahriak** [linkerd/linkerd2#6017] [Q&A]
Hello,

I am trying to install linkerd2 and linkerd-viz components using a single Helm install command. For that I have created a requirements file and included linkerd-viz as a dependency. I am aware that linkerd core components needs to be installed before linkerd-viz components can successfully start. Therefore, I have included a post install hook to simply restart the linkerd-viz components after the linkerd check command succeeds (kubectl rollout restart deploy -n linkerd-viz). Now the issue I am running into is, the proxy-init image is not being able to be pulled from the private registry after the post install hook executes the command. I have updated the _proxy_init.tpl file under the partials folder to point to the registry as well. I am not sure what I am missing..

Error message:
Failed to pull image "proxy-init:v1.3.9": rpc error: code = Unknown desc = Error response from daemon: pull access denied for proxy-init, repository does not exist or may require 'docker login': denied: requested access to the resource is denied

**Comment by kleimkuhler** [comment]
@shahriak When using a private registry you'll want to make sure you are overriding the default registry values in the charts. If installing linkerd2-viz first, you'll want to override it's [defaultRegistry](https://github.com/linkerd/linkerd2/blob/b53dc3b400e16e882a1f98538cd795f64e109700/viz/charts/linkerd-viz/values.yaml#L16) value: `--set defaultRegistry='private.registry/linkerd'`.

This is assuming that your linkerd2 chart installation is also pointing at the private registry. linkerd2 chart has a few additional registry fields for [proxy.image.name](https://github.com/linkerd/linkerd2/blob/19b31ee894e2450a1d8ea8f9399da617d13fd8aa/charts/linkerd2/values.yaml#L57), [proxyInit.image.name](https://github.com/linkerd/linkerd2/blob/19b31ee894e2450a1d8ea8f9399da617d13fd8aa/charts/linkerd2/values.yaml#L116-L117), and [controllerImage](https://github.com/linkerd/linkerd2/blob/19b31ee894e2450a1d8ea8f9399da617d13fd8aa/charts/linkerd2/values.yaml#L154). So the linkerd2 chart installation should pass the flag: `--set proxy.image.name='private.registry/linkerd/proxy',proxyInit.image.name='private.registry/linkerd/proxy-init',controllerImage='private.registry/linkerd/controller'`.
